pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -382,24 +380,27 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule, push *model.PushContex
 	}
 }
 
-func (a *v1beta1PolicyApplier) PortLevelSetting() map[uint32]*v1beta1.PeerAuthentication_MutualTLS {
-	return a.consolidatedPeerPolicy.PortLevelMtls
+func (a v1beta1PolicyApplier) PortLevelSetting() map[uint32]v1beta1.PeerAuthentication_MutualTLS_Mode {
+	return a.consolidatedPeerPolicy.PerPort
 }
 
-func (a *v1beta1PolicyApplier) GetMutualTLSModeForPort(endpointPort uint32) model.MutualTLSMode {
-	if a.consolidatedPeerPolicy.PortLevelMtls != nil {
-		if portMtls, ok := a.consolidatedPeerPolicy.PortLevelMtls[endpointPort]; ok {
-			return getMutualTLSMode(portMtls)
-		}
+func (a v1beta1PolicyApplier) GetMutualTLSModeForPort(endpointPort uint32) model.MutualTLSMode {
+	if portMtls, ok := a.consolidatedPeerPolicy.PerPort[endpointPort]; ok {
+		return getMutualTLSMode(portMtls)
 	}
 
-	return getMutualTLSMode(a.consolidatedPeerPolicy.Mtls)
+	return getMutualTLSMode(a.consolidatedPeerPolicy.Mode)
 }
 
 // getMutualTLSMode returns the MutualTLSMode enum corresponding peer MutualTLS settings.
 // Input cannot be nil.
-func getMutualTLSMode(mtls *v1beta1.PeerAuthentication_MutualTLS) model.MutualTLSMode {
-	return model.ConvertToMutualTLSMode(mtls.Mode)
+func getMutualTLSMode(mtls v1beta1.PeerAuthentication_MutualTLS_Mode) model.MutualTLSMode {
```

Comment:
super nit: why not just call `model.ConvertToMutualTLSMode` directly from PortLevelSettings?

Reply:
agree